pkg/products/cloudresources/reconciler.go

```diff
@@ -163,6 +163,21 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		return phase, err
 	}
 
+	// Check if STS Cluster, get STS role ARN addon parameter and pass ARN to Secret in CRO namespace
+	r.log.Info("checking if STS mode")
+	isSTS, err := sts.IsClusterSTS(ctx, client, r.log)
```

Comment:
I see we are also checking if the cluster is STS in rhmi controller, can you think of any better way of doing it? No strong opinion but I guess I would have hoped that there would be something in the RHMI CR that would indicate whether or not we are on STS.

Reply:
The RHMI CR does not have a property relating to STS. Currently, we're checking if the cluster has STS enabled by fetching the CloudCredential CR and checking if its mode is `manual`. Another approach which I just came up with is to call this endpoint and check if the path `aws.sts` exists 🤔